ui/addon/src/main/java/org/jboss/windup/ui/WindupUpdateDistributionCommand.java

```diff
@@ -0,0 +1,157 @@
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.context.UIValidationContext;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.windup.util.PathUtil;
+
+/**
+ * Provides a basic UI command updating the rules/migration-core and addons folders with the latest version.
+ *
+ * @author mbriskar
+ */
+public class WindupUpdateDistributionCommand implements UICommand
+{
+
+    @Inject
+    private DependencyResolver dependencyResolver;
+
+    @Override
+    public UICommandMetadata getMetadata(UIContext ctx)
+    {
+        return Metadata.forCommand(getClass()).name("Windup Update Distribution").description("Update the whole windup installation")
+                    .category(Categories.create("Platform", "Migration"));
+    }
+
+    @Override
+    public Result execute(UIExecutionContext context) throws Exception
+    {
+        //TODO:Make this to resolve distribution version (however there is no way to know what is current windup version)
+        if (!RulesetUpdateChecker.rulesetNeedUpdate(dependencyResolver))
+        {
+            return Results.fail("The ruleset is already in the most updated version.");
+        }
+        List<Coordinate> distributionVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder.create(CoordinateBuilder.create()
+                    .setGroupId("org.jboss.windup")
+                    .setArtifactId("windup-distribution")));
+        Coordinate latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - 1);
+        int i=0;
+        do{
+            i++;
+            latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - i);
+        }while(latestDistributionCoordinate.isSnapshot()); 
+        
+        Path windupRulesDir = PathUtil.getWindupRulesDir();
+        Path addonsDir = PathUtil.getWindupAddonsDir();
+        Path coreRulesPropertiesPath = windupRulesDir.resolve(RulesetUpdateChecker.RULESET_CORE_DIRECTORY);
+        // delete the previous rules and addons
+        deleteWholeDirectory(coreRulesPropertiesPath);
+        deleteWholeDirectory(addonsDir);
+        Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(CoordinateBuilder.create()
+                    .setGroupId(latestDistributionCoordinate.getGroupId()).setArtifactId(latestDistributionCoordinate.getArtifactId())
+                    .setVersion(latestDistributionCoordinate.getVersion()).setClassifier("offline").setPackaging("zip")));
+        FileResource<?> artifact = dependency.getArtifact();
+        ZipFile zipFile = new ZipFile(artifact.getFullyQualifiedName());
```

Comment:
Do we have a utility for handling zip files? If so, we probably want to use that here.